.github/workflows/security.yml

```diff
@@ -1,65 +1,60 @@
     runs-on: ${{ matrix.os }}
+    concurrency:
+      cancel-in-progress: true
+      group: security-analysis-${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
     strategy:
-      fail-fast: false
+      fail-fast: true
       matrix:
-        # Note: This workflow requires only the LATEST version of PHP
-        php: [ 8.2 ]
-        os: [ ubuntu-latest ]
+        os:
+          - ubuntu-latest
+        php-version:
+          - '8.2'
+        dependencies:
+          - locked
     steps:
-      - name: Set up PHP ${{ matrix.php }}
-        uses: shivammathur/setup-php@v2
-        with:
-          php-version: ${{ matrix.php }}
-          extensions: dom, sockets, grpc, curl
+      - name: 📦 Check out the codebase
+        uses: actions/checkout@v4.1.4
 
-      - name: Check Out Code
-        uses: actions/checkout@v4
+      - name: 🛠️ Setup PHP
+        uses: shivammathur/setup-php@2.30.4
         with:
-          fetch-depth: 1
+          php-version: ${{ matrix.php-version }}
+          extensions: none, ctype, dom, json, mbstring, phar, simplexml, tokenizer, xml, xmlwriter, sockets
+          ini-values: error_reporting=E_ALL
+          coverage: none
+
+      - name: 🛠️ Setup problem matchers
+        run: echo "::add-matcher::${{ runner.tool_cache }}/php.json"
 
-      - name: Get Composer Cache Directory
-        id: composer-cache
-        run: echo "::set-output name=dir::$(composer config cache-files-dir)"
+      - name: 🤖 Validate composer.json and composer.lock
+        run: composer validate --ansi --strict
 
-      - name: Cache Dependencies
-        uses: actions/cache@v3
+      - name: 🔍 Get composer cache directory
+        uses: wayofdev/gh-actions/actions/composer/get-cache-directory@v3.0.0
+
+      - name: ♻️ Restore cached dependencies installed with composer
+        uses: actions/cache@v4.0.2
         with:
-          path: ${{ steps.composer-cache.outputs.dir }}
-          key: php-${{ matrix.php }}-${{ runner.os }}-composer-${{ hashFiles('**/composer.lock') }}
-          restore-keys: php-${{ matrix.php }}-${{ runner.os }}-composer-
+          path: ${{ env.COMPOSER_CACHE_DIR }}
+          key: php-${{ matrix.php-version }}-composer-${{ matrix.dependencies }}-${{ hashFiles('composer.lock') }}
+          restore-keys: php-${{ matrix.php-version }}-composer-${{ matrix.dependencies }}-
 
-      - name: Install Composer Dependencies
-        run: composer install --prefer-dist --no-interaction
+      - name: 📥 Install "${{ matrix.dependencies }}" dependencies
+        uses: wayofdev/gh-actions/actions/composer/install@v3.0.0
+        with:
+          dependencies: ${{ matrix.dependencies }}
 
-      - name: Verify
+      - name: 🐛 Check installed packages using roave/security-advisories
         run: composer require --dev roave/security-advisories:dev-latest
+
+      - name: 🐛 Check installed packages for security vulnerability advisories
+        run: composer audit --ansi
```

Comment:
should we use `roave/security-advisories` if we use `composer audit`?

Reply:
Yes, you are correct; the previous approach wasn’t fully effective.

`composer audit` is specifically designed for CI workflows to perform security checks after dependencies are installed, which aligns with our process where we install package dependencies first and then run `composer audit`.

`roave/security-advisories` acts as a proactive tool by blocking the installation of packages with known vulnerabilities from the beginning. It's most effective when it’s included in `composer.json` as a development dependency and activates during `composer install` or `update` actions.

Currently, we do not add `roave/security-advisories` to our `require-dev` in order to keep our development dependencies minimal.

If we're looking to boost our security measures, integrating `roave/security-advisories` could provide an extra layer of protection by preventing the installation of vulnerable packages before they are even flagged by `composer audit`. For this to be effective, we would need to include this package in our `require-dev` dependencies.